cnf-certification-test/lifecycle/suite.go

```diff
@@ -584,23 +581,30 @@ func TestPodTolerationBypass(env *provider.TestEnvironment) {
 	testhelper.AddTestResultLog("Non-compliant", podsWithRestrictedTolerationsNotDefault, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
 func testStorageRequiredPods(env *provider.TestEnvironment) {
-	oc := clientsholder.GetClientsHolder()
-	var nonCompliantPods []*provider.Pod
-	for _, pod := range env.Pods {
-		for _, podOwner := range pod.ObjectMeta.GetOwnerReferences() {
-			if podOwner.Kind != statefulSet {
-				continue
-			}
-			statefulset, err := oc.K8sClient.AppsV1().StatefulSets(pod.Namespace).Get(context.TODO(), podOwner.Name, apiv1.GetOptions{})
-			if err != nil {
-				tnf.ClaimFilePrintf(err.Error())
+	var podsWithLocalStorage []string
+	for _, put := range env.Pods {
+		for pvIndex := range put.Spec.Volumes {
+			// Skip any nil persistentClaims.
+			if put.Spec.Volumes[pvIndex].PersistentVolumeClaim == nil {
 				continue
 			}
-			if statefulset.Spec.ServiceName == localStorage {
-				nonCompliantPods = append(nonCompliantPods, pod)
-				continue
+			// We have the list of pods/volumes/claims.
+			// Look through the storageClass list for a match.
+			for _, s := range env.StorageClasses {
+				if s.Name == put.Spec.Volumes[pvIndex].PersistentVolumeClaim.ClaimName {
```

Comment:
you are comparing the storage class name with the PVC name  these will not match. First, you need to retrieve the PVC object corresponding to put.Spec.Volumes[pvIndex].PersistentVolumeClaim.ClaimName and in the PVC you will be able to see the storage class. Looks like we already get the PVCs in env.PersistentVolumeClaims so you can get the PVC object from there

Reply:
The changes have been uploaded @edcdavid